packages/labs/context/src/lib/controllers/context-provider.ts

```diff
@@ -83,19 +83,50 @@ export class ContextProvider<T extends Context<unknown, unknown>>
     // The check on composedPath (as opposed to ev.target) is to cover cases
     // where the consumer is in the shadowDom of the provider (in which case,
     // event.target === this.host because of event retargeting).
-    if (ev.context !== this.context || ev.composedPath()[0] === this.host) {
+    const consumerHost = ev.composedPath()[0] as Element;
+    if (ev.context !== this.context || consumerHost === this.host) {
       return;
     }
     ev.stopPropagation();
-    this.addCallback(ev.callback, ev.subscribe);
+    this.addCallback(ev.callback, ev.subscribe, consumerHost);
+  };
+
+  public onProviderRequest = (
```

Comment:
really minor nit: do we want explicit `public` keyword?

Reply:
non blocking nit: Worth adding comments on the intent of this handler? Maybe write a little about what you explained to me in person.

Reply:
> really minor nit: do we want explicit public keyword?

It's a big odd, but I'm following local convention. Raised as a topic on the discord.

Good point on docs, added!